### PR TITLE
[events v2] Sample of how to check for event type attributes in the extended checker

### PR DESCRIPTION
**Pull request by wrwg**
Since this kind of check (where we look at each call) hasn't been done before, this PR demonstrates how the plumbing could look like.

### CODE REVIEW

**Review by davidiw — COMMENTED** [review]
Wow! Nice.

@lightmark , maybe your dreams can come true....

you can use this to get your options 2 and 3 without doing anything too fun upstream.

File: aptos-move/framework/src/extended_checks.rs [hunk 1]
```diff
@@ -67,6 +72,7 @@ impl<'a> ExtendedChecker<'a> {
                 self.check_and_record_resource_group_members(module);
                 self.check_and_record_view_functions(module);
                 self.check_entry_functions(module);
+                self.check_events(module);
```
Comment (lightmark) [inline aptos-move/framework/src/extended_checks.rs:75]: should we do both check and record?

**Comment by lightmark** [comment]
> > Thanks @wrwg . My understanding is that the check will be called when compiling the code but not will not be verified by verifier. So if someone bypass compilation it is still possible to emit `T` w/o `#[event]` attribute, right? Does this mean we have to either have similar logic in verifier or runtime native functions?
> 
> This check can be applied during module publishing: https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/aptos-vm/src/aptos_vm.rs#L994

So the final static check is at module publishing phase, basically repeat the logic of this PR?

**Comment by brmataptos** [comment]
Note this, however: https://github.com/aptos-labs/aptos-core/issues/9251, though it's not impossible that I'm mistaken.

**Comment by wrwg** [comment]
> > > Thanks @wrwg . My understanding is that the check will be called when compiling the code but not will not be verified by verifier. So if someone bypass compilation it is still possible to emit `T` w/o `#[event]` attribute, right? Does this mean we have to either have similar logic in verifier or runtime native functions?
> > 
> > 
> > This check can be applied during module publishing: https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/aptos-vm/src/aptos_vm.rs#L994
> 
> So the final static check is at module publishing phase, basically repeat the logic of this PR?

Yes, that is what we also discussed previously. However, I'm not convinced that we need to do any runtime check here. Keep in mind that runtime code verification is all about safety. So what specifically is unsafe about calling emit with a struct which does not have the event attribute? Just wondering.

**Comment by lightmark** [comment]
> So what specifically is unsafe about calling emit with a struct which does not have the event attribute? Just wondering.
I think it is not about safety but following our event emission protocol. So I finally didn't do any native function runtime check but only check that in module publishing request as what resource group did. I believe it reaches a balance between complexity, performance hit and protocol robustness.

**Comment by davidiw** [comment]
Closing out, as we now have the code being used in a functioning PR. https://github.com/aptos-labs/aptos-core/pulls?q=is%3Apr+is%3Aopen+events

**Pull request closed without merging**